### PR TITLE
Fix: Blurred images in attachment view on mobile devices

### PR DESCRIPTION
**Pull request by aswin-s**
### Details

This PR fixes issue an issue with image attachment view, where the images were blurred out once loaded. 

### Fixed Issues

$ https://github.com/Expensify/App/issues/5193

### Tests

1. Navigate to a conversation
2. Send an image
3. Tap on the image and zoom
4. Verify that the image shown is sharp and fits to screen width.

### QA Steps

1. Navigate to a conversation
2. Send an image
3. Tap on the image and zoom
4. Verify that the image shown is sharp and fits to screen.
5. Repeat above steps with images of different dimensions, especially large images like [this](https://user-images.githubusercontent.com/44479856/132889916-e46ac3dc-19ad-4cca-a033-ec13b8ac79ad.jpg)

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### Screenshots

Issue is present only on mobile devices.

#### iOS

image

https://user-images.githubusercontent.com/6880914/143115542-1fe0637c-171e-45ac-b94b-ea39eea84380.mp4

#### Android

Screenshot_1637700815

https://user-images.githubusercontent.com/6880914/143119438-d12e7d1d-0623-41ab-9f1d-920abbe8ca84.mp4

### CODE REVIEW

File: src/components/ImageView/index.native.js [hunk 1]
```diff
@@ -102,21 +144,14 @@ class ImageView extends PureComponent {
                         this.imageZoomScale = scale;
                     }}
                 >
-                    <ImageWithSizeCalculation
```
Comment (iwiznia) [inline src/components/ImageView/index.native.js:105]: hmmm I wonder if we should move all this logic to this component instead... what's the value of keeping both?

Comment (aswin-s) [reply src/components/ImageView/index.native.js:105]: `ImageWithSizeCalculation` component is used in Web & Desktop platforms as well and current implementation works just fine for those. Keeping the logic contained in ImageView native implementation, we basically keep the unnecessary dependencies out of web bundle & avoid webpack errors. If you still prefer to keep it in ImageSizeWithCalculation then it needs to have a native implementation which uses FastImage & RNImageSize.

Comment (iwiznia) [reply src/components/ImageView/index.native.js:105]: Oh ok ok, no, this is good.

**Comment by iwiznia** [comment]
Argh damn, sorry you got conflicts.

**Comment by aswin-s** [comment]
@iwiznia Resolved conflicts

File: src/components/ImageView/index.native.js [hunk 2]
```diff
@@ -43,7 +47,12 @@ class ImageView extends PureComponent {
 
     componentDidMount() {
         // Wait till animations are over to prevent stutter in navigation animation
-        InteractionManager.runAfterInteractions(() => this.calculateImageSize());
+        this.state.interactionPromise = InteractionManager.runAfterInteractions(() => this.calculateImageSize());
+    }
+
+    componentWillUnmount() {
+        if (!this.state.interactionPromise) { return; }
```
Comment (iwiznia) [inline src/components/ImageView/index.native.js:54]: I think our style guide says to not do this (ie: we should have return in a new line and the close bracket in another one.
Having said that, you can change this to: `this.state.interactionPromise && this.state.interactionPromise.cancel()`

Comment (aswin-s) [reply src/components/ImageView/index.native.js:54]: On that actually throws `Expected an assignment or function call and instead saw an expression.eslintno-unused-expressions` lint error.

Comment (aswin-s) [reply src/components/ImageView/index.native.js]: I changed it to return in new line.

**Review by iwiznia — APPROVED** [review]

**Comment by iwiznia** [comment]
Just realized I had to approve you as contributor for the workflow to run. Please ping me again to merge it when the workflow runs successfully.

**Comment by aswin-s** [comment]
@iwiznia All checks passed.

**Comment by iwiznia** [comment]
Awesome, can you clarify in the QA steps to do the tests with images of various sizes, especially with big image sizes.

**Comment by roryabraham** [comment]
As stated [over here](https://github.com/Expensify/App/pull/6535#issuecomment-987014288), I'm going to revert this PR because it caused a regression.

**Comment by roryabraham** [comment]
Damn, couldn't be reverted automatically 😭

**Comment by s77rt** [comment]
This PR is party responsible for a regression here https://github.com/Expensify/App/issues/15288 The `react-native-fast-image` lib is bugged.

File: src/components/ImageView/index.native.js [hunk 3]
```diff
@@ -41,6 +45,34 @@ class ImageView extends PureComponent {
         });
     }
 
+    componentDidMount() {
+        // Wait till animations are over to prevent stutter in navigation animation
+        this.state.interactionPromise = InteractionManager.runAfterInteractions(() => this.calculateImageSize());
```
Comment (s77rt) [inline src/components/ImageView/index.native.js:50]: @aswin-s Do you recall the production steps for the animation bug here? We are looking to remove `InteractionManager.runAfterInteractions`.

Comment (aswin-s) [reply src/components/ImageView/index.native.js:50]: Wow this was over 2 years ago. If I remember correctly trying to calculate image size while animating the modal was causing frozen frames on android device. But this component has gone through multiple refactors after this change and even react-native-fast-image was forked to improve load performance. So this optimisation might not be required anymore.

Comment (s77rt) [reply src/components/ImageView/index.native.js:50]: Thank you!

**Pull request merged**